### PR TITLE
I18N-1324 - Update TextUnitSearcherParameters to support Builder pattern

### PR DESCRIPTION
**Pull request by DarKhaos**
Updated TextUnitSearcherParameters to support Builder pattern in ThirdPartyTMSSmartling class

### CODE REVIEW

File: webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java [hunk 1]
```diff
@@ -927,14 +930,15 @@ private Long singularCount(
...
         .countTextUnitAndWordCount(
-            baseParams(
-                repositoryId,
-                localeTag,
-                skipTextUnitsWithPattern,
-                skipAssetsWithPathPattern,
-                true,
-                true,
-                null))
+            this.baseParams(
```
Comment (garionpin) [inline webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java:933]: Now that we have `TextUnitSearcherParameters.Builder`, could we have baseParams just have the common conditions preconfigured (rootlocale =false, used filter = Used, etc.) and have it return a builder - and everything that's different from call to call to use the native builder pattern?

It should also make calls like this a lot more obvious as to what params they're passing, once replaced with the builder:
```
this.baseParams(
                    repositoryId,
                    localeTag,
                    skipTextUnitsWithPattern,
                    skipAssetsWithPathPattern,
                    false,
                    false,
                    "%")
                .build())
```
--->
```
this.baseParams()
    .repositoryIds(repositoryId)
    .pluralFormsFiltered(False)
    # ... etc
    .build())
```

**Review by garionpin — COMMENTED** [review]
Added note re baseParams

File: webapp/src/main/java/com/box/l10n/mojito/service/tm/search/TextUnitSearcherParameters.java [hunk 2]
```diff
@@ -346,4 +346,272 @@ public Duration getAiTranslationExpiryDuration() {
...
+      this.locationUsage = locationUsage;
+      return this;
+    }
+
+    public Builder searchType(SearchType searchType) {
+      this.searchType = searchType;
+      return this;
+    }
+
+    public Builder repositoryIds(Long repositoryId) {
```
[the remark this answers is not in the document]
Comment (DarKhaos) [reply webapp/src/main/java/com/box/l10n/mojito/service/tm/search/TextUnitSearcherParameters.java]: Done

File: webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java (hunk 1 above)
Comment (DarKhaos) [reply webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java]: Done

**Review by garionpin — APPROVED** [review]

**Pull request merged**